### PR TITLE
Identity changes for AuthN work

### PR DESCRIPTION
**Pull request by HaoK**
Mostly just switching from ClaimsIdentity -> ClaimsPrincipal, AuthenticationScheme -> AuthenticationType everywhere

cc @divega @lodejard @Tratcher @blowdart

### CODE REVIEW

**Comment by HaoK** [comment]
Needed to make some changes to the Controllers to adapt to the new world (cc @rustd)

Logged in check change:
(old)`User.Identity.IsAuthenticated` => (new)`User.IsLoggedIn()` 

IsLoggedIn is a new Identity extension method on IPrincipal from Identity which looks for a ClaimsIdentity with the ApplicationCookie's authentication type]

**Comment by HaoK** [comment]
Renamed IsSignedIn => IsLoggedIn for consistency

**Comment by HaoK** [comment]
Removed old IClaimsIdentity extension methods and moved them to IPrincipal ...

```
User.Identity.GetUserId => User.GetUserId
User.Identity.GetUserName => User.GetUserName

File: Identity.sln [hunk 1]
```diff
@@ -28,6 +28,52 @@ Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNet.Identity.E
 EndProject
 Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNet.Identity.EntityFramework", "src\Microsoft.AspNet.Identity.EntityFramework\Microsoft.AspNet.Identity.EntityFramework.kproj", "{4490894C-3572-4E63-86F1-EE5105CE8A06}"
 EndProject
+Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "Microsoft.AspNet.Http", "..\HttpAbstractions\src\Microsoft.AspNet.Http\Microsoft.AspNet.Http.kproj", "{22071333-15BA-4D16-A1D5-4D5B1A83FBDD}"
+EndProject
```
Comment (Praburaj) [inline Identity.sln]: Revert this change. I guess you have included packages by source.

File: global.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-    "sources": ["src"]
+    "sources": ["src", "C:\\Github\\Security\\src", "C:\\Github\\HttpAbstractions\\src", "C:\\Github\\WebListener\\src", "C:\\Github\\Mvc\\src"]
```
Comment (Praburaj) [inline global.json]: revert this.

File: samples/IdentitySample.Mvc/LocalConfig.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
     "DefaultAdminPassword": "YouShouldChangeThisPassword1!",
     "Data": {
         "IdentityConnection": {
-            "Connectionstring": "Server=(localdb)\\mssqllocaldb;Database=IdentityMvc-1-7-15;Trusted_Connection=True;MultipleActiveResultSets=true"
+            "Connectionstring": "Server=(localdb)\\mssqllocaldb;Database=IdentityMvc-2-20-15-3;Trusted_Connection=True;MultipleActiveResultSets=true"
         }
```
Comment (Praburaj) [inline samples/IdentitySample.Mvc/LocalConfig.json]: nit: revert this change?

**Comment by Praburaj** [comment]
Looks good to me. :shipit:

**Comment by HaoK** [comment]
Merged 3aaa6283653a9d44b746767ec04fb7159553cb9e
